e2e/deployment.go

```diff
@@ -296,3 +297,151 @@ func (rnr *rookNFSResource) Do(action kubectlAction) error {
 
 	return nil
 }
+
+func waitForDeploymentUpdateScale(
+	c kubernetes.Interface,
+	ns,
+	deploymentName string,
+	scale *autoscalingv1.Scale,
+	timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err := wait.PollImmediate(poll, t, func() (bool, error) {
+		scaleResult, upsErr := c.AppsV1().Deployments(ns).UpdateScale(context.TODO(),
+			deploymentName, scale, metav1.UpdateOptions{})
+		if upsErr != nil {
+			if isRetryableAPIError(upsErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment UpdateScale %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error update scale deployment %s: %w", deploymentName, upsErr)
+		}
+		if scaleResult.Spec.Replicas != scale.Spec.Replicas {
+			e2elog.Logf("scale result not matching for deployment %s, desired scale %d, got %d",
+				deploymentName, scale.Spec.Replicas, scaleResult.Spec.Replicas)
+
+			return false, fmt.Errorf("error scale not matching in deployment %s: %w", deploymentName, upsErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update scale deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForDeploymentUpdate(
+	c kubernetes.Interface,
+	ns,
+	deploymentName string,
+	deployment *appsv1.Deployment,
+	timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err := wait.PollImmediate(poll, t, func() (bool, error) {
+		_, upErr := c.AppsV1().Deployments(ns).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+		if upErr != nil {
+			if isRetryableAPIError(upErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment Update %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error updating deployment %s: %w", deploymentName, upErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForClusterNameUpdate(c kubernetes.Interface, ns, deploymentName string, timeout int) error {
```

Comment:
keep the function generic pass container name (s) and flag the set as input.

Reply:
How? May be I'm missing the point, according to me this is generic enough to play around a deployment?

Reply:
This will only update clusterName argument what if later we want to add some other argument?

Reply:
So what do you want the function schema to look like?

Reply:
making hardcoded values as input looks okay to be generic?

```go
// containers check if slice contains string.
func contains(s []string, e string) bool {
	for _, a := range s {
		if a == e {
			return true
		}
	}

	return false
}

....
for i := range containers {
		if contains(containers[i].Name == containerName) {
                       for _,ak:=range containers[i].Args{
                              if string.Contains(ak,key){
                               // do replacement with key and value
                              // break
                         }
                        // if they key is not present append it to the args
			containers[i].Args = append(containers[i].Args, fmt.Sprintf("--%s=%s", key,value))
			deployment.Spec.Template.Spec.Containers[i].Args = containers[i].Args
		}
	}
```

Reply:
requested changes not present in the latest update